applications/grandpa_verification.md

```diff
@@ -0,0 +1,580 @@
+
+### Team members
+
+- Mario Altimari
+- Luis Alberto Díaz Díaz
+- Sparsa Roychowdhury
+
+### Contact
+
+- **Contact Name:** Mario Altimari
+- **Contact Email:** <marioaltimari@mlabs.city>
+- **Website:** <https://www.mlabs.city>
+
+### Legal Structure
+
+- **Registered Address:** Address of your registered legal entity, if available. Please keep it in a single line. MLabs LTD, 1st-floor Citibase Millbank Tower,21-24 Millbank, LONDON SW1P 4QP, United Kingdom
+- **Registered Legal Entity:** MLabs LTD
+
+### Team's experience
+
+- Mario has witnessed the evolution of the web from its early days to the present, as an IT expert with more than two decades of experience. He has worked on various projects with different roles using many technologies.
+- Luis is a mathematician, in 2020 he began a master of science with focus on formal verification, he will graduate this year with a thesis in modal logic. He has been working for MLabs since January 2022 for multiple projects Cardano based
+- Sparsa have been working as Formal Verification Engineer since 2022. He has extensive experience in formally verifying hardware and software. He developed new models and methodologies during his PhD that can be used to formally verify real-time systems with concurrency.
+
+Mario Altimari applied for a Web3 Foundation grant before: TuxedoDapp by Mlabs
+
+### Team Code Repos
+
+- Mario Altimari [https://github.com/AltiMario](https://github.com/AltiMario)
+- Luis Alberto Díaz Díaz [https://github.com/Luis-omega](https://github.com/Luis-omega)
+- Sparsa Roychowdhury publications <https://dblp.org/pid/259/1497.html>
+
+### Team LinkedIn Profiles (if available)
+
+- Mario Altimari <https://www.linkedin.com/in/altimario/>
+- Luis Alberto Díaz Díaz <https://www.linkedin.com/in/luis-alberto-d%C3%ADaz-d%C3%ADaz-4aa882226/>
+- Sparsa Roychowdhury <https://www.linkedin.com/in/sparsa-roychowdhury-phd-6336b831/>
+
+## Development Status :open_book:
+
+We have had conversations with the W3F team, planning to start a collaboration in this area
+
+## Development Roadmap :nut_and_bolt:
+
+### Estimate for **_The Model Checker TLC_**
+
+|Description of the algorithm|Encoding effort|Verification effort|Combined|
+|----------------------------|---------------|-------------------|--------|
+|Initiate GRANDPA|2h|2h|4h|
+|Play GRANDPA-round|24h|24h|48h|
+|Derive-Primary|1h|1h|2h|
+|Best Final Candidate|4h|4h|8h|
+|GRANDPA Ghost|4h|4h|8h|
+|Best pre-Vote Candidate|2h|2h|4h|
+|Attempt to Finalize at Round|4h|4h|8h|
+|Finalizable|8h|8h|16h|
+|Process Catchup Request|8h|8h|16h|
+|Process Catchup Response|8h|8h|16h|
+|TOTAL|||130h
+
+- **Total Estimated Duration:** 1 month
+- **Full-Time Equivalent (FTE):**  1.5 FTE
+- **Total Costs:** $13000 (USD)
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Verification | We will provide encoding and property verification using TLA+ as described in detail|
```

Comment:
Can you please add documentation for the techniques and abstractions used for modelling as a deliverable. Ideally with in-line comments in the TLA+ spec.